RPSWithErrorMessages/Program.cs

```diff
@@ -0,0 +1,80 @@
+﻿using System;
+
+namespace RockPaperScissors
+{
+    class Program
+    {
+        public static void Main()
+        {
+            //asks for user input
+            Console.WriteLine("Enter your move:");
+            string hand1 = Console.ReadLine().ToLower();
+
+            //if player enters invalid hand, throws exception and asks for valid input
+            while(hand1 != "rock"&&hand1 != "paper"&&hand1 != "scissors"){
+                try{
+                    
+                    if (hand1 != "rock"&&hand1 != "paper"&&hand1 != "scissors")
+                    {
+                        throw new Exception ("Invalid move");
+                    }
+                }
+                catch(Exception) {
+                     
+                    Console.WriteLine("Invalid move entered please try again");
+                    hand1 = Console.ReadLine().ToLower();
```

Comment:
This is will work, but its a little less elegant the way you have it written. You are doing the check 2 times for valid input. Once for they condition to throw an exception, and once for the condition to check if there is a valid hand so you can keep asking for input. 

Also, the way you have it, a do/While loop would fit better.